Epoch  0 Epoch Loss :  0.006647911840751336
Training loss:  1.37218022358661e-05  Training accuracy:  0.7036851851851852
Validation loss:  0.00012633886425618459  Validation accuracy:  0.6995
Testing loss:  7.932239603112446e-05  Testing accuracy:  0.6904
Epoch  1 Epoch Loss :  0.004850145897469886
Training loss:  1.2188953869891891e-05  Training accuracy:  0.7981296296296296
Validation loss:  0.00011346427437020052  Validation accuracy:  0.794
Testing loss:  7.29777604073813e-05  Testing accuracy:  0.779
Epoch  2 Epoch Loss :  0.005006740734963797
Training loss:  1.2379645761168706e-05  Training accuracy:  0.8088333333333333
Validation loss:  0.00011349019216841826  Validation accuracy:  0.8025
Testing loss:  7.337041838928474e-05  Testing accuracy:  0.798
Epoch  3 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
/home/sadbhawna/Desktop/cs6910_assignment1/Feedforward_Neural_Network.py:61: RuntimeWarning: invalid value encountered in divide
  return exponential_A / exponential_A.sum(axis=0, keepdims=True)
Epoch  4 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  5 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  6 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  7 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  8 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  9 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  10 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Epoch  11 Epoch Loss :  nan
Training loss:  nan  Training accuracy:  0.1
Validation loss:  nan  Validation accuracy:  0.1
Testing loss:  nan  Testing accuracy:  0.1
Traceback (most recent call last):
  File "/home/sadbhawna/Desktop/cs6910_assignment1/train.py", line 63, in <module>
    fit(layer_architecture, X_train, Y_train, X_val, Y_val, X_test, Y_test, epochs=55, activation='relu', loss = 'cross_entropy', optimizer='rmsprop', weight_ini = 'He Normal',learning_rate=0.01, batch=128, weight_decay=0.0, epsilon=1e-6, project="cs23d014_assignment_1")
  File "/home/sadbhawna/Desktop/cs6910_assignment1/train.py", line 29, in fit
    rmsprop(layer_architecture, X_train, Y_train, X_val, Y_val, X_test, Y_test, epochs=epochs, activation=activation, loss=loss, weight_ini = weight_ini, learning_rate=learning_rate, beta=beta, batch=batch, epsilon=epsilon, weight_decay=weight_decay, project="cs23d014_assignment_1")
  File "/home/sadbhawna/Desktop/cs6910_assignment1/optimizers.py", line 258, in rmsprop
    wandb.log({'confusion_matrix': wandb.plot.confusion_matrix(probs=None, y_true=y_true_class_indx,preds=preds_class_indx,class_names=labels)})
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 420, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 371, in wrapper_fn
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 361, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 1828, in log
    self._log(data=data, step=step, commit=commit)
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 1596, in _log
    self._partial_history_callback(data, step, commit)
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/wandb_run.py", line 1468, in _partial_history_callback
    self._backend.interface.publish_partial_history(
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/interface/interface.py", line 558, in publish_partial_history
    data = history_dict_to_json(run, data, step=user_step, ignore_copy_err=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/data_types/utils.py", line 52, in history_dict_to_json
    payload[key] = val_to_json(
                   ^^^^^^^^^^^^
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/data_types/utils.py", line 156, in val_to_json
    art.add(val, key)
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/artifacts/artifact.py", line 1375, in add
    with self.new_file(name) as f:
  File "/home/sadbhawna/anaconda3/lib/python3.11/contextlib.py", line 144, in __exit__
    next(self.gen)
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/sdk/artifacts/artifact.py", line 1122, in new_file
    with util.fsync_open(path, mode, encoding) as f:
  File "/home/sadbhawna/anaconda3/lib/python3.11/contextlib.py", line 144, in __exit__
    next(self.gen)
  File "/home/sadbhawna/anaconda3/lib/python3.11/site-packages/wandb/util.py", line 1517, in fsync_open
    os.fsync(f.fileno())
KeyboardInterrupt